CLI NPE on Ubuntu 20.04 LTS

Same binary works ok on my Ubuntu 18.4.4 LTS but fails on Ubuntu 20.04 LTS,
CLI bin version: 2.0.0-RC1, master(11.6.2020)
```
kec@ubukec:~/idp/playground$ cp ~/helidon /usr/bin/helidon
kec@ubukec:~/idp/playground$ sudo chmod 777 /usr/bin/helidon
kec@ubukec:~/idp/playground$ helidon init
Exception in thread "main" java.lang.NullPointerException
	at java.util.regex.Matcher.getTextLength(Matcher.java:1770)
	at java.util.regex.Matcher.reset(Matcher.java:416)
	at java.util.regex.Matcher.<init>(Matcher.java:253)
	at java.util.regex.Pattern.matcher(Pattern.java:1133)
	at java.util.Formatter.parse(Formatter.java:2700)
	at java.util.Formatter.format(Formatter.java:2655)
	at java.util.Formatter.format(Formatter.java:2609)
	at java.lang.String.format(String.java:2897)
	at io.helidon.build.util.Style.render(Style.java:259)
	at io.helidon.build.util.SystemLogWriter.toStyled(SystemLogWriter.java:156)
	at io.helidon.build.util.SystemLogWriter.write(SystemLogWriter.java:129)
	at io.helidon.build.util.Log.log(Log.java:266)
	at io.helidon.build.cli.harness.CommandContext$ExitAction.exit(CommandContext.java:183)
	at io.helidon.build.cli.harness.CommandContext$ExitAction.lambda$run$1(CommandContext.java:162)
	at java.util.Optional.ifPresentOrElse(Optional.java:203)
	at io.helidon.build.cli.harness.CommandContext$ExitAction.run(CommandContext.java:156)
	at io.helidon.build.cli.harness.CommandRunner.execute(CommandRunner.java:123)
	at io.helidon.build.cli.impl.Main.main(Main.java:36)
kec@ubukec:~/idp/playground$ cd myproject/
kec@ubukec:~/idp/playground/myproject$ helidon build
Exception in thread "main" java.lang.NullPointerException
	at java.util.regex.Matcher.getTextLength(Matcher.java:1770)
	at java.util.regex.Matcher.reset(Matcher.java:416)
	at java.util.regex.Matcher.<init>(Matcher.java:253)
	at java.util.regex.Pattern.matcher(Pattern.java:1133)
	at java.util.Formatter.parse(Formatter.java:2700)
	at java.util.Formatter.format(Formatter.java:2655)
	at java.util.Formatter.format(Formatter.java:2609)
	at java.lang.String.format(String.java:2897)
	at io.helidon.build.util.Style.render(Style.java:259)
	at io.helidon.build.util.SystemLogWriter.toStyled(SystemLogWriter.java:156)
	at io.helidon.build.util.SystemLogWriter.write(SystemLogWriter.java:129)
	at io.helidon.build.util.Log.log(Log.java:266)
	at io.helidon.build.cli.harness.CommandContext$ExitAction.exit(CommandContext.java:183)
	at io.helidon.build.cli.harness.CommandContext$ExitAction.lambda$run$1(CommandContext.java:162)
	at java.util.Optional.ifPresentOrElse(Optional.java:203)
	at io.helidon.build.cli.harness.CommandContext$ExitAction.run(CommandContext.java:156)
	at io.helidon.build.cli.harness.CommandRunner.execute(CommandRunner.java:123)
	at io.helidon.build.cli.impl.Main.main(Main.java:36)

```

When I have printed out the Exception returning `null` as `getMessage`
```
kec@ubukec:~/idp/helidon-build-tools/helidon-cli/impl/target$ ./helidon init
java.lang.ExceptionInInitializerError
	at com.oracle.svm.core.hub.ClassInitializationInfo.initialize(ClassInitializationInfo.java:290)
	at java.lang.Class.ensureInitialized(DynamicHub.java:499)
	at io.helidon.build.util.MavenCommand.mavenExecutable(MavenCommand.java:107)
	at io.helidon.build.util.MavenCommand.mavenHome(MavenCommand.java:129)
	at io.helidon.build.util.MavenCommand.installedVersion(MavenCommand.java:142)
	at io.helidon.build.util.MavenCommand.assertRequiredMavenVersion(MavenCommand.java:163)
	at io.helidon.build.cli.impl.CommandRequirements.requireMinimumMavenVersion(CommandRequirements.java:77)
	at io.helidon.build.cli.impl.InitCommand.execute(InitCommand.java:145)
	at io.helidon.build.cli.harness.CommandRunner.doExecuteCommand(CommandRunner.java:76)
	at java.util.Optional.ifPresentOrElse(Optional.java:201)
	at io.helidon.build.cli.harness.CommandRunner.doExecuteCommandName(CommandRunner.java:59)
	at java.util.Optional.ifPresentOrElse(Optional.java:201)
	at io.helidon.build.cli.harness.CommandRunner.doExecute(CommandRunner.java:49)
	at java.util.Optional.ifPresentOrElse(Optional.java:203)
	at io.helidon.build.cli.harness.CommandRunner.execute(CommandRunner.java:42)
	at io.helidon.build.cli.harness.CommandRunner.execute(CommandRunner.java:108)
	at io.helidon.build.cli.harness.CommandRunner.execute(CommandRunner.java:122)
	at io.helidon.build.cli.impl.Main.main(Main.java:36)
Caused by: java.lang.RuntimeException: Unable to find Java installation, please set JAVA_HOME
	at io.helidon.build.util.Constants.javaHome(Constants.java:64)
	at io.helidon.build.util.FileUtils.<clinit>(FileUtils.java:51)
	at com.oracle.svm.core.hub.ClassInitializationInfo.invokeClassInitializer(ClassInitializationInfo.java:350)
	at com.oracle.svm.core.hub.ClassInitializationInfo.initialize(ClassInitializationInfo.java:270)
	... 17 more
kec@ubukec:~/idp/helidon-build-tools/helidon-cli/impl/target$ whereis java
java: /usr/bin/java /usr/share/java /usr/lib/jvm/graalvm-ce-java11-20.1.0/bin/java /usr/share/man/man1/java.1.gz
kec@ubukec:~/idp/helidon-build-tools/helidon-cli/impl/target$ ll -la /usr/bin/java
lrwxrwxrwx 1 root root 22 kvě  4 13:12 /usr/bin/java -> /etc/alternatives/java*
```

--- a/helidon-cli/harness/src/main/java/io/helidon/build/cli/harness/CommandContext.java
+++ b/helidon-cli/harness/src/main/java/io/helidon/build/cli/harness/CommandContext.java
@@ -180,6 +180,16 @@ public final class CommandContext {
                 if (Style.isStyled(message)) {
                     Log.info(message);
                 } else {
+                    if (message == null) {
+                        if (error.getMessage() != null) {
+                            message = error.getMessage();
+                        } else if (error.getCause() != null) {
+                            message = error.getCause().getMessage();
+                        } else {
+                            message = "Unknown error";
+                            error.printStackTrace();
+                        }
+                    }
                     Log.log(level, error, message);
                 }
                 if (Log.isVerbose()) {

--- a/helidon-linker/src/test/java/io/helidon/linker/LinkerTest.java
+++ b/helidon-linker/src/test/java/io/helidon/linker/LinkerTest.java
@@ -120,7 +120,7 @@ class LinkerTest {
         Path mainAppJar = FileUtils.assertFile(appDir.resolve(mainJarName));
         assertReadOnly(mainAppJar);
         Path appLibDir = FileUtils.assertDir(appDir.resolve("libs"));
-        for (Path file : FileUtils.listFiles(appLibDir, name -> true)) {
+        for (Path file : FileUtils.listFiles(appLibDir, (path, attrs) -> attrs.isRegularFile())) {
             assertReadOnly(file);
         }
     }

--- a/utils/src/main/java/io/helidon/build/util/FileUtils.java
+++ b/utils/src/main/java/io/helidon/build/util/FileUtils.java
@@ -180,7 +180,7 @@ public final class FileUtils {
     public static List<Path> listFiles(Path directory, Predicate<String> fileNameFilter, int maxDepth) {
         try {
             return Files.find(assertDir(directory), maxDepth, (path, attrs) ->
-                    attrs.isRegularFile() && fileNameFilter.test(path.getFileName().toString())
+                    fileNameFilter.test(path.getFileName().toString())
             ).collect(Collectors.toList());
         } catch (IOException e) {
             throw new UncheckedIOException(e);

--- a/utils/src/main/java/io/helidon/build/util/FileUtils.java
+++ b/utils/src/main/java/io/helidon/build/util/FileUtils.java
@@ -34,6 +34,7 @@ import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.BiPredicate;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
@@ -170,6 +171,17 @@ public final class FileUtils {
     }
 
     /**
+     * List all files in the given directory that match the given filter. Does not recurse.
+     *
+     * @param directory The directory.
+     * @param pathFilter The filter.
+     * @return The normalized, absolute file paths.
+     */
+    public static List<Path> listFiles(Path directory, BiPredicate<Path, BasicFileAttributes> pathFilter) {
+        return listFiles(directory, pathFilter, 1);
+    }
+
+    /**
      * List all files in the given directory that match the given filter, recursively if maxDepth > 1.
      *
      * @param directory The directory.
@@ -178,10 +190,20 @@ public final class FileUtils {
      * @return The normalized, absolute file paths.
      */
     public static List<Path> listFiles(Path directory, Predicate<String> fileNameFilter, int maxDepth) {
+        return listFiles(directory, (path, attrs) -> fileNameFilter.test(path.getFileName().toString()), maxDepth);
+    }
+
+    /**
+     * List all files in the given directory that match the given filter, recursively if maxDepth > 1.
+     *
+     * @param directory The directory.
+     * @param pathFilter The filter.
+     * @param maxDepth The maximum recursion depth.
+     * @return The normalized, absolute file paths.
+     */
+    public static List<Path> listFiles(Path directory, BiPredicate<Path, BasicFileAttributes> pathFilter, int maxDepth) {
         try {
-            return Files.find(assertDir(directory), maxDepth, (path, attrs) ->
-                    fileNameFilter.test(path.getFileName().toString())
-            ).collect(Collectors.toList());
+            return Files.find(assertDir(directory), maxDepth, pathFilter).collect(Collectors.toList());
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }

--- a/utils/src/main/java/io/helidon/build/util/MavenCommand.java
+++ b/utils/src/main/java/io/helidon/build/util/MavenCommand.java
@@ -25,11 +25,13 @@ import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Predicate;
+import java.util.jar.JarFile;
 import java.util.stream.Collectors;
 
 import static io.helidon.build.util.AnsiConsoleInstaller.isHelidonChildProcess;
@@ -145,12 +147,31 @@ public class MavenCommand {
             if (jars.isEmpty()) {
                 throw new IllegalStateException(MAVEN_CORE_PREFIX + "* not found in " + libDir);
             }
-            final String fileName = jars.get(0).getFileName().toString();
-            final String versionStr = fileName.substring(MAVEN_CORE_PREFIX.length(), fileName.length() - JAR_SUFFIX.length());
+            final Path jarFile = jars.get(0);
+            final String fileName = jarFile.getFileName().toString();
+            final String versionStr = jarVersion(jarFile)
+                    .orElse(fileName.substring(MAVEN_CORE_PREFIX.length(), fileName.length() - JAR_SUFFIX.length()));
             final MavenVersion version = MavenVersion.toMavenVersion(versionStr);
             MAVEN_VERSION.set(version);
         }
         return MAVEN_VERSION.get();
+    }
+
+    /**
+     * Find implementation version from manifest file.
+     *
+     * @param jarFilePath Jar file to look for version in
+     * @return version or empty optional
+     */
+    public static Optional<String> jarVersion(Path jarFilePath) {
+        try {
+            return Optional.of(new JarFile(jarFilePath.toFile())
+                    .getManifest()
+                    .getMainAttributes()
+                    .getValue("Implementation-Version"));
+        } catch (Throwable t) {
+            return Optional.empty();
+        }
     }
 
     /**

